### ISSUE TITLE
content type header missing on 11ty dev server

### ISSUE THREAD

**Issue by btrem** [11ty/eleventy-dev-server#46]
### Operating system

Ubuntu 22.04.1

### Eleventy

2.0.0-canary.33

### Describe the bug

11ty dev server is not consistently sending a content type header.

For my home page, it sends `text/html; charset=utf-8`. For other pages, the content-type header is missing.

### Reproduction steps

1. clone test repo https://github.com/btrem/11ty_dev_server_test
2. run `npm install`
3. run `npm start`
4. view dev server home page with a dev tools extension (e.g., Firefox Dev Edition)
5. follow link to any page listed on home page with dev tools extension

### Expected behavior

The content type header should be sent for all pages.

### Reproduction URL

https://github.com/btrem/11ty_dev_server_test

**Comment by sguter90** [comment]
I think this isn't directly a dev server problem.  
As soon as permalinks are set without ``/index.html`` suffix the generated files do not have a file extension.  
An example from the provided repository:  
The permalink for posts is configured to ``/{{ page.date | njkDate('YYYY/MM/DD') }}-{{ slug | slug }}``.  
``place-called-africa.md`` will create the file ``_site/2022/02/28-place-called-africa``  

The devserver tries to identifiy the mime type but can't because of the missing file extension.
This leads to a missing ``Content-Type`` header.

This output behaviour is also consistent with eleventy version 1.0.2 even if the [11ty docs](https://www.11ty.dev/docs/permalinks/#changing-the-output-location) describe that it doesn't matter if ``index.html`` is added to the permalink or not.

Adding ``index.html`` can workaround the problem.  
In your specific example:  
``/{{ page.date | njkDate('YYYY/MM/DD') }}-{{ slug | slug }}/index.html``

**Comment by btrem** [comment]
My permalinks don't end with `index.html`. Is there a way to set a default content type, like one can do with Apache?

**Comment by btrem** [comment]
It looks like #47 would resolve this issue.

**Comment by sguter90** [comment]
No, not directly I‘m sorry. While implementing I‘ve searched for existing issues which I could fix within this PR and found this issue.  

I decided to not change behaviour here because default configurations of production webservers will also not send the correct Conent-Type in this case. Nginx for example falls back to text/plain. (per default)

Instead I think either eleventy itself should output an folder with an index.html or adjust the docs to provide the correct way how to generate permalinks.  

I know that permalinks without that index.html look prettier.
I think you can add a middleware which matches URLs without extensions or some other matching logic and than set the Content-Type by yourself. (Works if #47 is merged)

**Comment by btrem** [comment]
The docs already explain how to set a permalink. But there is no "correct way...to generate permalinks." While Apache defaults to `text/plain`, it is easy to change it. I imagine that the same is true of Nginx. It is also possible with the server that came with 11ty v 1.x, which I could change with a configuration in `.eleventy.js`.

As such, it should be possible to do the same with 11ty-dev-server. Will PR #47 allow a similar config change?

**Comment by sguter90** [comment]
While doing a research on this topic I came across this issue: https://github.com/11ty/eleventy/issues/1599  
I really have to say that I didn't read the docs far enough to see that warning:
> While index.html is optional, it is a [Common Pitfall](https://www.11ty.dev/docs/pitfalls/) to leave off the trailing slash! If you forget it, the browser may attempt to download the file instead of displaying it (unless you’ve done some extra work to set up your Content-Type headers correctly).

So the solution is to just add a trailing slash:  
before: ``/{{ page.date | njkDate('YYYY/MM/DD') }}-{{ slug | slug }}``  
after: ``/{{ page.date | njkDate('YYYY/MM/DD') }}-{{ slug | slug }}/``  

To answer your question anyway: Yes you can set kind of default Content-Type.
But you don't need #47 for  this. You can just move the ``middleware`` part which was previously set via ``setBrowserSyncConfig`` to ``setServerOptions``:
```js
    eleventyConfig.setServerOptions({
        middleware: [
            function (req, res, next) {
                if (/^[^.]+$/.test(req.url)) {
                    res.setHeader('Content-Type', 'text/html; charset=utf-8');
                }
                next();
            }
        ]
    })
```

The problem which is solved in #47 is that if a mime-type is detected the Content-Type header in always set (even if already defined in a middleware)
In your case the mime-type can not be detected so setting the Content-Type already works.

**Comment by btrem** [comment]
Ah, I didn't know I could port my old setBrowserSyncConfig code to setServerOptions. That resolves my problem. Thanks for your time.
